src/storage.rs

```diff
@@ -262,7 +293,69 @@ impl<'a, T, A, D> Join for &'a mut Storage<T, A, D>
     }
 }
 
+#[cfg(feature="serialize")]
+impl<T, A, D> serde::Serialize for Storage<T, A, D> where
+    T: Component + serde::Serialize,
+    A: Deref<Target = Allocator>,
+    D: Deref<Target = MaskedStorage<T>>,
+{
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        use bitset::BitSetLike;
+        use serde::ser::SerializeStruct;
+
+        // Serializes the storage in a format of PackedData<T>
+        let (bitset, storage) = self.open();
+        let mut structure = serializer.serialize_struct("PackedData", 2)?;
+        let mut components: Vec<&T> = Vec::new();
+        let mut offsets: Vec<u32> = Vec::new();
+        for index in bitset.iter() {
+            offsets.push(index);
+            unsafe {
+                components.push(storage.get(index));
+            }
+        }
+
+        structure.serialize_field("offsets", &offsets)?;
+        structure.serialize_field("components", &components)?;
+        structure.end()
+    }
+}
+
+#[cfg(feature="serialize")]
+#[derive(Debug, Serialize, Deserialize)]
+/// Structure of packed components with offsets of which entities they belong to.
+/// Offsets define which entities the components correspond to, based on a list of entities
+/// the packed data is sent in with.
 ///
+/// If the list of entities is all entities in the world, then the offsets in the 
+/// packed data are the indices of the entities.
+pub struct PackedData<T> {
+    /// List of components.
+    pub components: Vec<T>,
+    /// Offsets used to get entities which correspond to the components.
+    pub offsets: Vec<u32>,
+}
+
+#[cfg(feature="serialize")]
+impl<T> PackedData<T> {
+    /// Shifts all offsets in the packed data to match a new base.
+    ///
+    /// Useful if you want to merge some components ontop of the current world:
+    /// `packed.rebase( ... /* amount of entities in world */ );`
+    ///
+    /// If the base is higher than the offset then the function will return the index of that offset.
+    pub fn rebase(&mut self, base: u32) -> Option<usize> {
+        for index in 0..self.offsets.len() {
+            if self.offsets[index] < base {
+                return Some(index);
+            }
+            self.offsets[index] -= base;
```

Comment:
I'm thinking I might just remove the rebase stuff since you could just use the entity list instead (and it is less weird because it won't get wrapping weirdness if the base is higher than the offsets).